### PR TITLE
dev-python/pyphen: EAPI=7, add py3.7+py3.8, test, l10n IUSE for dict inclusion

### PR DESCRIPTION
**Pull request by wimmuskee**
* EAPI 7
* Python 3.7 + Python 3.8
* python_test phase
* l10n IUSE for hyphenation file inclusion

### CODE REVIEW

**Comment by wimmuskee** [comment]
hmm, while making a PR for upstream, encountered a bug
upstream has symlink-based fallback to hyph_<lang>.dic, while the myspell packages don't make those

**Comment by wimmuskee** [comment]
Too much patching was needed in upstream code to facilitate using system dictionaries without having fallback problems. Changed the patch to at least be able to set which of pyphen's dictionaries are included.

File: dev-python/pyphen/pyphen-0.9.4-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+HOMEPAGE="https://github.com/Kozea/Pyphen"
+SRC_URI="https://github.com/Kozea/Pyphen/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2+ LGPL-2+ MPL-1.1"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+# some dictionaries are used for testing
+REQUIRED_USE="test? ( l10n_en l10n_fr l10n_hu l10n_nl )"
+RESTRICT="!test? ( test )"
```
[the remark this answers is not in the document]
Comment (sbraz) [reply dev-python/pyphen/pyphen-0.9.4-r1.ebuild]: This is expected: https://devmanual.gentoo.org/general-concepts/dependencies/#test-dependencies

File: dev-python/pyphen/Manifest [hunk 2]
```diff
@@ -1 +1,2 @@
 DIST Pyphen-0.9.4.tar.gz 1092404 BLAKE2B 9bd96c0d0dc7a7ccb9bebaf7e864ec0212956ac910830d9a3286fc9569d9b044f7aedbc3d807260e5b5e1ccbe84ffa22c686680ebf7352cd23d6ed2755f7d92e SHA512 b48dcf02c5cdf5d7ee6d6968644958af9f31fba046612016d9ae1b8cfe44d438fe769592961f93f3342cc3f3af2dfc6cd40bbfe1be67e59de9a352c33f102917
+DIST pyphen-0.9.4.tar.gz 1100291 BLAKE2B f7b6470327d35cc04c161e45986ae551e9ce2d83e0c06609fcc774eac768a5f19093d9804dd3347ff87402e1b3ef1304d72ab7261c6f9fb7642b8c9b5c69e7f7 SHA512 f008f1563c1ec195096d7d61be4fc226facf04191fadd2da79b03da8f4308b096b3259246c915fd1a75f7a4d54544970f55289ddaf94a8db40ed3a8b091558a5
```
[the remark this answers is not in the document]
Comment (wimmuskee) [reply dev-python/pyphen/Manifest]: The current tree 0.9.4 is from pypi (not Github) and does not include tests.

File: dev-python/pyphen/pyphen-0.9.4-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+PYTHON_COMPAT=( python{2_7,3_6,3_7,3_8} )
```
Comment (Zlogene) [inline dev-python/pyphen/pyphen-0.9.4-r1.ebuild]: do not keep py27

File: dev-python/pyphen/pyphen-0.9.4-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+DISTUTILS_USE_SETUPTOOLS=bdepend
+PLOCALES="af af_ZA bg bg_BG ca cs cs_CZ da da_DK de de_AT de_CH de_DE el el_GR en en_GB en_Latn_GB en_Latn_US en_US es et et_EE fr gl hr hr_HR hu hu_HU is it it_IT lt lt_LT lv lv_LV nb nb_NO nl nl_NL nn nn_NO pl pl_PL pt pt_BR pt_Latn_BR pt_Latn_PT pt_PT ro ro_RO ru ru_RU sk sk_SK sl sl_SI sr sr_Latn sv te te_IN uk uk_UA zu zu_ZA"
```
Comment (Zlogene) [inline dev-python/pyphen/pyphen-0.9.4-r1.ebuild]: this is simply obsolete (we do not use PLOCALES anymore)

File: dev-python/pyphen/pyphen-0.9.4-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+inherit distutils-r1 l10n
```
Comment (Zlogene) [inline dev-python/pyphen/pyphen-0.9.4-r1.ebuild]: drop l10n

File: dev-python/pyphen/pyphen-0.9.4-r1.ebuild (hunk 3 above)
Comment (wimmuskee) [reply dev-python/pyphen/pyphen-0.9.4-r1.ebuild]: Check, no reverse dependencies on python2.7-only apps.

**Comment by wimmuskee** [comment]
@titanofold, good call on the 0.9.5 version. In the first commit in this PR I tried to make use of the system myspell dictionaries (and 0.9.5 only updates dicts).
The tests run on python3.8 (altough not mentioned upstream), Also, py36 and py37 are also not set in upstream test (and current tree version is set on 3.6).
Shortened the commit message. Also, making a not to change some other commits in open PR's.

File: dev-python/pyphen/pyphen-0.9.4-r1.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (wimmuskee) [reply dev-python/pyphen/pyphen-0.9.4-r1.ebuild]: I thought this was expected as well, restrict test if the test USE is not set.

File: dev-python/pyphen/pyphen-0.9.4-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+done
+
+src_prepare() {
+	default
+	l10n_find_plocales_changes "${S}/pyphen/dictionaries" "hyph_" '.dic'
+
+	# removing dictionary files that are not specified through L10N
+	for lang in ${PLOCALES}; do
+		if [[ ${#lang} -eq 2 ]] && use !l10n_${lang}; then
+			rm pyphen/dictionaries/hyph_${lang}*
```
[the remark this answers is not in the document]
Comment (wimmuskee) [reply dev-python/pyphen/pyphen-0.9.4-r1.ebuild]: done

File: dev-python/pyphen/pyphen-0.9.4-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+		fi
+	done
+}
+
+python_test() {
+	py.test -vv test.py || die "Tests fail with ${EPYTHON}"
```
[the remark this answers is not in the document]
Comment (wimmuskee) [reply dev-python/pyphen/pyphen-0.9.4-r1.ebuild]: changed py.test to pytest

File: dev-python/pyphen/pyphen-0.9.4-r1.ebuild (hunk 5 above)
Comment (wimmuskee) [reply dev-python/pyphen/pyphen-0.9.4-r1.ebuild]: Could you elaborate on that please? I use the l10n for the l10n_find_plocales_changes function to see if there are changed dictionaries between versions.

File: dev-python/pyphen/pyphen-0.9.5.ebuild [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+done
+
+src_prepare() {
+	default
+	l10n_find_plocales_changes "${S}/pyphen/dictionaries" "hyph_" '.dic'
+
+	# removing dictionary files that are not specified through L10N
+	for lang in ${PLOCALES}; do
+		if [[ ${#lang} -eq 2 ]] && use !l10n_${lang}; then
+			rm pyphen/dictionaries/hyph_${lang}* || die "Removing dictionary file(s) failed"
```
Comment (titanofold) [inline dev-python/pyphen/pyphen-0.9.5.ebuild]: Split this long line (generally try to avoid going over 80 characters).

Comment (wimmuskee) [reply dev-python/pyphen/pyphen-0.9.5.ebuild]: done

**Comment by thesamesam** [comment]
(Think this is obsolete now? Please ping us in proxy-maint if something stalls for ages.)

**Pull request closed without merging**